### PR TITLE
Simplify tests and remove uncessary config files everest/mathfunc

### PR DESCRIPTION
**Pull request by StephanDeHoop**
**Issue**
Resolves [#9210](https://github.com/equinor/ert/issues/9210)

**Approach**
Simplify tests and remove unnecessary config files in everest/math_func. Since (https://github.com/equinor/ert/pull/9357) was a complete overkill will close that PR and work on this one. ~For the life of me I am struggling to remove either `config_remove_run_path` or `config_fm_failure` since they are completely the same except the `fail` argument. However, when I try to read `config_remove_run_path` and replace the `config.forward_model` with `["toggle_failure --fail simulation_2"]` and set the `log_level` to `info` the tests keeps failing and I don't understand why.~

**NOTE**: Test coverage should not have changed with this PR, only the number of config files that we check into our repository has been drastically reduced and some tests should be a bit more explicit in what they are testing. 

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'pytest tests/ert/unit_tests -n logical -m "not integration_test"'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/everest/test_detached.py [hunk 1]
```diff
@@ -48,7 +48,12 @@
 @pytest.mark.fails_on_macos_github_workflow
 @pytest.mark.xdist_group(name="starts_everest")
 async def test_https_requests(copy_math_func_test_data_to_tmp):
-    everest_config = EverestConfig.load_file("config_minimal_slow.yml")
+    everest_config = EverestConfig.load_file("config_minimal.yml")
+    everest_config.install_jobs.append(
+        InstallJobConfig(name="sleep", source="jobs/SLEEP")
+    )
+    everest_config.forward_model.append("sleep --sleep 10")
+
```
Comment (oyvindeide) [inline tests/everest/test_detached.py:56]: Looks good, but could probably be simplified even more, as we dont really need the full math func case to test this, for example:

```suggestion
async def test_https_requests(tmp_path, monkeypatch, min_config):
    monkeypatch.chdir(tmp_path)
    with open("sleep_job", "w", encoding="utf-8") as fout:
        fout.write("EXECUTABLE sleep")
    with open("config.yml", "w") as fout:
        yaml.dump(min_config | {
            "install_jobs": [{"name": "sleep", "source": "sleep_job"}],
            "forward_model": ["sleep --sleep 10"],
        }, fout)

    everest_config = EverestConfig.load_file("config.yml")
```

 where: min_config is a fixture in: tests/everest/test_everlint.py, which can be moved to a shared `conftest.py`
that way we could probably delete the `SLEEP` job and `sleep.py`, as it does not really make sense to have that checked in as a part of an example.

Can also keep as it, but suggest removing `sleep.py` and `jobs/SLEEP` regardless, as they are not relevant for an acual case, it is just for testing.

The other parts of the config is not really relevant for this test anyway.

**Review by oyvindeide — COMMENTED** [review]
This looks like a good direction! Removes a lot of duplication, and makes maintenance easier! I had a comment to one of the tests, we might be able to make some of the tests even more targeted.

File: tests/everest/test_detached.py (hunk 1 above)
Comment (StephanDeHoop) [reply tests/everest/test_detached.py:56]: Great suggestion, thanks :)!

**Comment by StephanDeHoop** [comment]
Will squash commits after review.

File: tests/everest/test_everlint.py [hunk 2]
```diff
@@ -13,6 +13,26 @@
...
+        max: 0.1
+        variables:
+          - { name: test, initial_guess: 0.1 }
+    objective_functions:
+      - {name: my_objective}
+    config_path: .
+    """)
+    )
+
+
```
Comment (verveerpj) [inline tests/everest/test_everlint.py:35]: This thing is already in `conftest.py`, I think? Does it need to be here? Is it duplicated now?

**Review by verveerpj — APPROVED** [review]
Looks good to me, but I have one question you may need to resolve before merging?

File: tests/everest/test_everlint.py (hunk 2 above)
Comment (StephanDeHoop) [reply tests/everest/test_everlint.py:35]: Ah, I completely missed that, we don't need it indeed :)! Thanks for spotting.

**Pull request merged**